charlotte_core/src/acpi/rsdp.rs

```diff
@@ -0,0 +1,106 @@
+use core::str;
+
+const RSDP_SIGNATURE: [u8; 8] = *b"RSD PTR ";
+
+const RDSP_V1_LEN: usize = 20; // Length of the RSDP for version 1 (20 bytes)
+
+/// Contains information about the RSDP (Root System Description Pointer).
+#[repr(C, packed)]
+#[derive(Copy, Clone)]
+pub struct Rsdp {
+    signature: [u8; 8],
+    checksum: u8,
+    oem_id: [u8; 6],
```

Comment:
maybe add an extra field with the oem_id as a parsed string.

Reply:
Yeah, I don't want to touch the struct fields themselves because it should ideally match the structure of the table in memory perfectly, but I modified the getter methods for these fields to return a `&str` instead of a `u8` slice.